contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,356 @@
+
+* Pods do not know how much local storage is available to them.
+* Pods cannot request “guaranteed” local storage.
+* Local storage is a “best-effort” resource 
+* Pods can get evicted due to other pods filling up the local storage during which time no new pods will be admitted, until sufficient storage has been reclaimed
+
+## Persistent Local Storage
+Distributed filesystems and databases are the primary use cases for persistent local storage due to the following factors:
+
+* Performance: On cloud providers, local SSDs give better performance than remote disks.
+* Cost: On baremetal, in addition to performance, local storage is typically cheaper and using it is a necessity to provision distributed filesystems.
+
+Distributed systems often use replication to provide fault tolerance, and can therefore tolerate node failures. However, data gravity is preferred for reducing replication traffic and cold startup latencies.
+
+# Design Overview
+
+A node’s local storage can be broken into primary and secondary partitions.  Primary partitions are shared partitions that can provide ephemeral local storage.  The two supported primary partitions are:
+
+### Root
+ This partition holds the kubelet’s root directory (`/var/lib/kubelet` by default) and `/var/log` directory. This partition may be shared between user pods, OS and Kubernetes system daemons. This partition can be consumed by pods via EmptyDir volumes, container logs, image layers and container writable layers. Kubelet will manage shared access and isolation of this partition. This partition is “ephemeral” and applications cannot expect any performance SLAs (Disk IO for example) from this partition.
+
+### Runtime 
+This is an optional partition which runtimes can use for overlay filesystems. Kubelet will attempt to identify and provide shared access along with isolation to this partition.
+
+All other partitions are exposed as persistent volumes. The PV interface allows for varying storage configurations to be supported, while hiding specific configuration details to the pod.  Applications can continue to use their existing PVC specifications with minimal changes to request local storage.
+
+# User Workflows
+
+### Alice manages a deployment and requires “Guaranteed” ephemeral storage
+
+1. Kubelet running across all nodes will identify primary partition and expose capacity and allocatable for the primary “root” partition. The runtime partition is an implementation detail and is not exposed outside the node.
+```yaml
+apiVersion: v1
+kind: Node
+metadata:
+  name: foo
+status:
+  Capacity: 
+    Storage: 100Gi
+  Allocatable:
+    Storage: 90Gi
+```
+2. Alice adds a “Storage” requirement to her pod as follows
+```yaml
+apiVersion: v1
+kind: pod
+metadata:
+ name: foo
+spec:
+ containers:
+  name: fooc
+  resources:
+   limits:
+    storage-logs: 500Mi
+    storage-overlay: 1Gi
```

Comment:
The main issue is there are no boundaries defined today in terms of local storage, so pods have free will to use as much storage as they like, whether its the writable layer, logs or emptydir volumes.  So we would like to impose capacity boundaries for all 3 areas.  

The challenge here is usability vs implementation.  From a user's point of view, they may not know or care about differentiating the 3 areas.  But from an implementation standpoint, since it is possible to have the optional runtime partition, hard limits can only be enforced if there are separate limits for each.